pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CloseResourceRule.java

```diff
@@ -214,8 +215,15 @@ private TypeNode getTypeOfVariable(ASTVariableDeclaratorId var) {
     }
 
     private TypeNode getDeclaredTypeOfVariable(ASTVariableDeclaratorId var) {
-        ASTLocalVariableDeclaration localVar = (ASTLocalVariableDeclaration) var.getParent().getParent();
-        return localVar.getTypeNode(); // note: can be null, if type is inferred (var)
+        JavaNode exprOrDecl = var.getParent().getParent();
```

Comment:
Isn't there already `ASTVariableDeclaratorId::getTypeNode` that does the same thing as this entire function ?

Reply:
Found it https://docs.pmd-code.org/apidocs/pmd-java/7.0.0-SNAPSHOT/net/sourceforge/pmd/lang/java/ast/ASTVariableDeclaratorId.html#getTypeNode()

Reply:
:+1: it works, thanks for the suggestion!